bigtop-deploy/puppet/manifests/cluster.pp

```diff
@@ -72,6 +72,10 @@
     master => ["spark-master"],
     worker => ["spark-worker"],
   },
+  spark-on-yarn => {
+    worker => ["spark-on-yarn", "spark-slave"],
+    client => ["spark-client"],
+  },
```

Comment:
Small typo above.  `spark-slave` should be `spark-yarn-slave` (see [init.pp](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/modules/spark/manifests/init.pp#L27)).

Also, while it's probably inconsequential, `spark-yarn-slave` isn't really a worker -- it just provides the yarn shuffle jar for spark.  So I think the above block should read:

```diff
+  spark-on-yarn => {
+    worker => ["spark-on-yarn"],
+    client => ["spark-client"],
+    library => ["spark-yarn-slave"],
+  },